### PR TITLE
Add polynomial_decay and piecewise_decay

### PR DESCRIPTION
**Pull request by jacquesqiao**
project: https://github.com/PaddlePaddle/Paddle/issues/7769
This PR will go on after switch_op https://github.com/PaddlePaddle/Paddle/pull/8031 is done.

### CODE REVIEW

File: python/paddle/v2/fluid/learning_rate_decay.py [hunk 1]
```diff
@@ -123,3 +123,54 @@ def inverse_time_decay(learning_rate,
...
     return learning_rate / (1 + decay_rate * div_res)
+
+
+def polynomial_decay(learning_rate,
+                     global_step,
+                     decay_steps,
+                     end_learning_rate=0.0001,
+                     power=1.0,
+                     cycle=False):
+    """Applies inverse time decay to the initial learning rate.
```
Comment (abhinavarora) [inline python/paddle/v2/fluid/learning_rate_decay.py]: This should be `Applies polynomial decay to the initial learning rate.`

Comment (jacquesqiao) [reply python/paddle/v2/fluid/learning_rate_decay.py]: ok~

File: python/paddle/v2/fluid/learning_rate_decay.py [hunk 2]
```diff
@@ -123,3 +126,98 @@ def inverse_time_decay(learning_rate,
...
+    """
+    if not isinstance(global_step, Variable):
+        raise ValueError("global_step is required for inverse_time_decay.")
+
+    if cycle:
+        div_res = layers.ceil(x=(global_step / decay_steps))
+        zero_var = layers.fill_constant(shape=[1], dtype='float32', value=0.0)
+        one_var = layers.fill_constant(shape=[1], dtype='float32', value=1.0)
+
+        with layers.Switch() as switch:
```
Comment (QiJune) [inline python/paddle/v2/fluid/learning_rate_decay.py]: So, our switch operator only support scalar condition now? And our if-else operator support vector condition.

File: python/paddle/v2/fluid/learning_rate_decay.py [hunk 3]
```diff
@@ -123,3 +126,98 @@ def inverse_time_decay(learning_rate,
...
+    else:
+        global_step = min(global_step, decay_steps)
+    decayed_learning_rate = (learning_rate - end_learning_rate) *
+                      (1 - global_step / decay_steps) ^ power +
+                      end_learning_rate
+    ```
+    Args:
+        learning_rate: A scalar float32 value or a Variable. This
+          will be the initial learning rate during training
+        global_step: A Variable that record the training step.
```
Comment (QiJune) [inline python/paddle/v2/fluid/learning_rate_decay.py]: So, where is the global_step variable created?

Comment (jacquesqiao) [reply python/paddle/v2/fluid/learning_rate_decay.py]: should be create outside optimizer. like
```python
    global_step = fluid.layers.create_global_var(shape=[1], value=0, dtype='float32', force_cpu=True)
    sgd_optimizer = fluid.optimizer.SGD(
        learning_rate=fluid.learning_rate_decay.exponential_decay(
            learning_rate=0.0001,
            global_step=global_step,
            decay_steps=100000,
            decay_rate=0.5,
            staircase=True),
        global_step=global_step)
```

File: python/paddle/v2/fluid/learning_rate_decay.py (hunk 2 above)
Comment (jacquesqiao) [reply python/paddle/v2/fluid/learning_rate_decay.py]: yes, the current switch operator only supports scalar condition.

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 4]
```diff
@@ -975,6 +976,36 @@ def less_than(x, y, cond=None, **ignored):
     return cond
 
 
+def equal(x, y, cond=None, **ignored):
```
Comment (QiJune) [inline python/paddle/v2/fluid/layers/control_flow.py]: Equal is a element-wise operator. And it also can be override in Python.
```
def __eq__(self):
    ...
```

File: python/paddle/v2/fluid/learning_rate_decay.py (hunk 3 above)
Comment (QiJune) [reply python/paddle/v2/fluid/learning_rate_decay.py]: If global_step variable is force cpu, does decay_steps also force cpu?

File: python/paddle/v2/fluid/layers/control_flow.py (hunk 4 above)
Comment (QiJune) [reply python/paddle/v2/fluid/layers/control_flow.py]: Or we can try to refine it in next PR
